### PR TITLE
made changes for correcting shift operators in tfhe sol

### PR DESCRIPTION
**Pull request by jatZama**
there is still some error that I do not understand where it comes from in the case of shift operators with : (uintX,euint8) signature. Codegen works but when running tfheOperations test with hardhat it tells me : 
```
TypeError: Member "shl" not unique after argument-dependent lookup in type(library TFHE).
  --> examples/tests/TFHETestSuite3.sol:10:25:
   |
10 |         euint8 result = TFHE.shl(aProc, bProc);
   |                         ^^^^^^^^
```

Which does not make sens for me since this function is defined only once with the (uint8,euint8) signature inside TFHE.sol

EDIT : this was solved by removing the shl/shr operators with (uintX,euitn8) signatures. Because solidity does not support this type of overloading :  https://forum.soliditylang.org/t/no-unique-declaration-found-when-overloading-functions-with-integers-of-different-sizes/1308 

In conclusion this PR removes the shift operators with a second operand greater than euint8/uint8 types.
By adding a `shiftOperator` key in the two shift operators objects, it implements the following operators : 
`shr(euintX,euint8)`, `shl(euintX,euint8)`, `shr(euintX,uint8)` and `shl(euintX,uint8)`, whenre `X` is `8`, `16` or `32`.

### CODE REVIEW

**Comment by aquint-zama** [comment]
@cla-bot check

**Comment by immortal-tofu** [comment]
You should clean your PR. It seems you have integrated previously merged commit concerning package.json and docs.

File: codegen/templates.ts [hunk 1]
```diff
@@ -473,7 +556,7 @@ function tfheAsEboolUnaryCast(bits: number): string {
     }
 
     function not(ebool a) internal pure returns (ebool) {
-        return asEbool(not(asEuint8(a)));
+        return asEbool(and(not(asEuint8(a)), asEuint8(1)));
```
Comment (david-zk) [inline codegen/templates.ts:559]: Hmmm this change is from previous commit, did you branch from the current master when making this PR?
